datafusion/physical-expr/src/aggregate/count.rs

```diff
@@ -76,6 +85,109 @@ impl Count {
     }
 }
 
+/// An accumulator to compute the counts of [`PrimitiveArray<T>`].
+/// Stores values as native types, and does overflow checking
+///
+/// Unlike most other accumulators, COUNT never produces NULLs. If no
+/// non-null values are seen in any group the output is 0. Thus, this
+/// accumulator has no additional null or seen filter tracking.
+#[derive(Debug)]
+struct CountGroupsAccumulator {
+    /// Count per group (use i64 to make Int64Array)
+    counts: Vec<i64>,
```

Comment:
I do not have any good answer for that 🤔

Reply:
Should I change it to u64?

Reply:
I think we should change it to u64 if possible :)

Reply:
So the reason this count is `i64` is because the output of `COUNT` is defined to be `DataType::Int64` and thus by using an i64 the output can be created directly. I will add a comment to clarify

```sql
❯ create table foo as values (1);
0 rows in set. Query took 0.004 seconds.
❯ select arrow_typeof(count(*)) from foo;
+-------------------------------+
| arrow_typeof(COUNT(UInt8(1))) |
+-------------------------------+
| Int64                         |
+-------------------------------+
1 row in set. Query took 0.004 seconds.
```

Reply:
f9eaa688cf

Reply:
Yeah - I think we should create an issue to make `COUNT` return `DataType::UInt64`